tests/foreman/api/test_virt_who_config.py

```diff
@@ -0,0 +1,408 @@
+"""Test for virt-who configure API
+
+:Requirement: Virtwho-configure
+
+:CaseLevel: Acceptance
+
+:CaseComponent: API
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:CaseAutomation: notautomated
+
+:Upstream: No
+"""
+
+from robottelo.decorators import run_only_on, stubbed, tier1
+from robottelo.test import TestCase, APITestCase
+
+
+class VirtWhoConfigAPI(APITestCase):
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_vm_create(self):
+        """ Register a vm on virt-who hypervisor
+
+        :id: 701b22b8-fe92-4dd3-8d7d-c7b5efb7281b
+
+        :steps:
+            1. Create new VM in a supported hypervisor
+            2. Wait until the next virt-who report comes in.
+            3. Verify the VM is reported to satellite
+
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_vm_create(self):
+        """ Register a vm on non-virt-who hypervisor
+
+        :id: abecc851-2475-4455-9c12-63a73fcb09bb
+
+        :steps:
+            1. Verify there are no virt-who reports reported if there is no
+               change in guest-host mapping in hypervisor
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_config_update(self):
+        """Update a config, verify it it changed and redploy it
+
+        :id: 6b2cc2c3-959b-468b-9865-0f01decd2249
+
+        :steps:
+            1. Update a config, verify it it changed and redploy it
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_postive_config_intervals(self):
+        """ Verify intervals Intervals
+
+        :id: 7e1bb498-4e63-44e8-be97-83e07601f56d
+
+        :steps:
+            1. Create virt-who configuration, set reporting interval to 1 hour.
+            2. Verify Virt-who  interval is 1 hour in config file.
+            3. Verify a report is sent every hour
+            4. Repeat for each supported interval.
+        """
+
+
+class VirtWhoConfigRoleApiTests(APITestCase):
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_role_manager(self):
+        """virt-who Manager allowed functions
+
+        :id: 4164bdde-f532-480c-b41e-747e87cf7d05
+        :steps:
+            1. Create a user with ONLY the virt-who manager role.
+            2. Verify the user can create virt-who configurations
+            3. Verify the user can edit an existing virt-who configuration
+            4. Verify the user can delete a virt-who configuration
+            5. Verify the user can see reporting info via the dashboard widget.
```

Comment:
Same here, some steps can be turned in to `expectedresults`  (apply this to all other stubs)